### PR TITLE
E2e CI job use memory storage for etcd

### LINKED ISSUE #2619: e2e: use memory storage for etcd
## Feature Request

This can speed up e2e testing in CI jobs and avoid `etcdserver: request timed out` error due to high disk load.

ref: https://github.com/kubernetes-sigs/kind/issues/845

### PR DESCRIPTION
**Pull request by shonge**
### What problem does this PR solve? 
fix #2619 

### What is changed and how does it work?
Add an option environment `ETCD_STORAGE_TYPE` , default is `disk`, if set it as `memory` will
running etcd use memory storage.
Reference:
https://github.com/kubernetes-sigs/kind/issues/845#issuecomment-529154066

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

Code changes

 - Has CI related scripts change

Side effects

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
None
```

### CODE REVIEW

File: ci/e2e_kind.groovy [hunk 1]
```diff
@@ -199,6 +199,7 @@ pipeline {
                 export DELETE_NAMESPACE_ON_FAILURE=${params.DELETE_NAMESPACE_ON_FAILURE}
                 export ARTIFACTS=${ARTIFACTS}
                 export GINKGO_NO_COLOR=y
+                export ETCD_STORAGE_TYPE=memory
```
Comment (shonge) [inline ci/e2e_kind.groovy:202]: I found this is not working, how to fix it?

File: hack/e2e.sh [hunk 2]
```diff
@@ -338,51 +339,85 @@ kubeadmConfigPatches:
...
+    else
+        if [[ "$KIND_DATA_HOSTPATH" != "none" ]]; then
+            if [ ! -d "$KIND_DATA_HOSTPATH" ]; then
+                echo "error: '$KIND_DATA_HOSTPATH' is not a directory"
+                exit 1
+            fi
+            local hostWorkerPath="${KIND_DATA_HOSTPATH}/control-plane"
+            test -d $hostWorkerPath || mkdir $hostWorkerPath
+            # set custom KIND_DATA_HOSTPATH
+            cat <<EOF >>$tmpfile
```
Comment (cofyc) [inline hack/e2e.sh:410]: this part can be simplified like this:

1)  add `extraMounts:` if `[[ "$KIND_ETCD_DATADIR" != "none" || "$KIND_DATA_HOSTPATH" != "none"   ]]`
2) add mounts for `/mnt/disks/`  if KIND_DATA_HOSTPATH != "none"
3) add mounts for `/var/lib/etcd` if KIND_ETCD_DATADIR  != "none"

Comment (cofyc) [reply hack/e2e.sh:410]: btw, please revert format changes in other places, focus one thing in one PR. you can submit another PR to change the format.

Comment (shonge) [reply hack/e2e.sh]: update

**Pull request merged**